R/QFeatures-constructors.R

```diff
@@ -138,3 +286,239 @@ QFeatures <- function(..., assayLinks = NULL) {
+        sampleMap = ans@sampleMap,
+        metadata = ans@metadata,
+        assayLinks = al)
+}
+
+.readQFeatures2 <- function(assayData, colAnnotation,
+                            batchCol, channelCol, suffix = NULL,
+                            sep = "", removeEmptyCols = FALSE,
+                            fnames, verbose = TRUE) {
+    if (missing(batchCol))
+        stop("Please provide a 'batchCol'.")
+    if (missing(channelCol))
+        stop("Please provide a 'channelCol'.")
+    ## Check the batch column name
+    if (!identical(make.names(batchCol), batchCol))
+        stop("'batchCol' is not a syntactically valid column name. ",
+             "See '?make.names' for converting the column names to ",
+             "valid names, e.g. '", batchCol, "' -> '",
+             make.names(batchCol), "'")
+    colData <- as.data.frame(colAnnotation)
+    ## Get the column contain the expression data
+    ecol <- unique(colData[, channelCol])
+    ## Get the sample suffix
+    if (is.null(suffix))
+        suffix <- ecol
+    ## Create the SummarizedExperiment object
+    if (verbose) message("Loading data as a 'SummarizedExperiment' object")
+    se <- readSummarizedExperiment(assayData, ecol)
+    if (missing(fnames)) {
+        rownames(se) <- paste0("PSM", seq_len(nrow(se)))
+    } else {
+        stopifnot(fnames %in% names(rowData(se)))
+        rownames(se) <- rowData(se)[, fnames]
+    }
+    ## Check the link between colData and se
+    mis <- !rowData(se)[, batchCol] %in% colData[, batchCol]
+    if (any(mis)) {
+        warning("Missing metadata. The features are removed for ",
+                paste0(unique(rowData(se)[mis, batchCol]), collapse = ", "))
+        se <- se[!mis, ]
+    }
+    ## Split the SingleCellExperiment object by batch column
+    if (verbose) message("Splitting data based on '", batchCol, "'")
+    se <- .splitSE(se, f = batchCol)
+    ## Clean each element in the data list
+    for (i in seq_along(se)) {
+        ## Add unique sample identifiers
+        colnames(se[[i]]) <- paste0(names(se)[[i]], sep, suffix)
+        ## Remove the columns that are all NA
+        if (removeEmptyCols) {
+            sel <- colSums(is.na(assay(se[[i]]))) != nrow(se[[i]])
+            se[[i]] <- se[[i]][, sel]
+        }
+    }
+    if (verbose) message("Formatting sample metadata (colData)")
+    ## Create the colData
+    cd <- DataFrame(row.names = unlist(lapply(se, colnames)))
+    rownames(colData) <- paste0(colData[, batchCol], sep, suffix)
+    cd <- cbind(cd, colData[rownames(cd), ])
+    ## Store the data as a QFeatures object
+    if (verbose) message("Formatting data as a 'QFeatures' object")
+    QFeatures(experiments = se, colData = cd)
+}
+
+
+
+##' @title  Read DIA-NN output as a QFeatures objects
+##'
+##' @description
+##'
+##' This function takes the output tables from DIA-NN and converts them
+##' into a `QFeatures` object.
+##'
+##' @param colData A `data.frame` or any object that can be coerced to
```

Comment:
`colAnnot` instead of `colData` for consistency with `readQFeatures()`?

Reply:
ok